File: Editorial/0101-0200/0120_Triangle/TypeScript/1_Dynamic_Programming.ts
function minimumTotal(triangle: number[][]): number {
    for (let row = 1; row < triangle.length; row++) {
        for (let col = 0; col <= row; col++) {
            let smallestAbove = Number.MAX_VALUE;
            if (col > 0) {
                smallestAbove = triangle[row - 1][col - 1];
            }
            if (col < row) {
                smallestAbove = Math.min(smallestAbove, triangle[row - 1][col]);
            }
            triangle[row][col] += smallestAbove;
        }
    }
    return Math.min(...triangle[triangle.length - 1]);
}

File: Editorial/0101-0200/0120_Triangle/CSharp/1_Dynamic_Programming.cs
public class Solution {
    public int MinimumTotal(IList<IList<int>> triangle) {
        for (int row = 1; row < triangle.Count; row++) {
            for (int col = 0; col <= row; col++) {
                int smallestAbove = int.MaxValue;
                if (col > 0) {
                    smallestAbove = triangle[row - 1][col - 1];
                }

                if (col < row) {
                    smallestAbove =
                        Math.Min(smallestAbove, triangle[row - 1][col]);
                }

                int path = smallestAbove + triangle[row][col];
                triangle[row][col] = path;
            }
        }

        return triangle[triangle.Count - 1].Min();
    }
}
